kotlin/viewregistry/src/main/java/com/squareup/viewregistry/AlertScreen.kt

```diff
@@ -32,10 +32,28 @@ data class AlertScreen(
   }
 
   sealed class Event {
-    data class ButtonClicked(
-      val button: Button
-    ) : Event()
+    data class ButtonClicked(val button: Button) : Event()
 
     object Canceled : Event()
   }
+
+  override fun equals(other: Any?): Boolean {
+    if (this === other) return true
+    if (javaClass != other?.javaClass) return false
+
+    other as AlertScreen
+
+    return buttons == other.buttons &&
+        message == other.message &&
+        title == other.title &&
+        cancelable == other.cancelable
+  }
+
+  override fun hashCode(): Int {
+    var result = buttons.hashCode()
+    result = 31 * result + message.hashCode()
+    result = 31 * result + title.hashCode()
+    result = 31 * result + cancelable.hashCode()
+    return result
```

Comment:
Really wish kotlin stdlib had a helper for this. I think it has a hidden intrinsic one somewhere already.